db/entrypoint.sh

```diff
@@ -46,6 +48,9 @@ if [ "$1" = 'postgres'  ]; then
 
     WAL_LEVEL=archive
     ARCHIVE_MODE=on
+    
+    echo "log_timezone = $DEFAULT_TIMEZONE" >> $PGDATA/postgresql.conf
```

Comment:
I guess the `$PGDATA` folder should be a mounted volume, so data inside it are persistent. So it seems that everytime the container restart it will add two new lines inside the `postgresql.conf` file. Am I right ? If true this is not a correct behavior, but I don't know a correct workaround for this.

Reply:
yes, right. this fixes errors:

```invalid value for parameter "log_timezone": "localtime"```
```invalid value for parameter "TimeZone": "localtime"```
```configuration file "/var/lib/postgresql/data/postgresql.conf" contains errors```

Reply:
Ok, but this is not ok if our container just append lines to a persistent configuration file at each restart.
I tried to reproduce the errors you show, but it seems that it didn't happen if you mount `/etc/localtime` inside the container like we do [on our `docker-compose.yml` file](https://github.com/mattermost/mattermost-docker/blob/master/docker-compose.yml#L10)

Reply:
Oups I just see that you can remove those two lines since you add this : 
```
sed -i "s/log_timezone =.*$/log_timezone = $DEFAULT_TIMEZONE/g" $PGDATA/postgresql.conf
sed -i "s/timezone =.*$/timezone = $DEFAULT_TIMEZONE/g" $PGDATA/postgresql.conf
```

Reply:
On RancherOS if there is not this line there will be errors that I sent above. Delete echo "log_timezone = $DEFAULT_TIMEZONE" >> $PGDATA/postgresql.conf ?

Reply:
I think the `sed` commands you add should fix it. Can you try without the two following `echo` instructions please ?
```
echo "log_timezone = $DEFAULT_TIMEZONE" >> $PGDATA/postgresql.conf
echo "timezone = $DEFAULT_TIMEZONE" >> $PGDATA/postgresql.conf
```

If there is still errors, we'll have to investigate on another way to fix it.

Reply:
RancherOS not work, if you delete :(

`echo "log_timezone = $ DEFAULT_TIMEZONE" >> $ PGDATA / postgresql.conf`
`echo "timezone = $ DEFAULT_TIMEZONE" >> $ PGDATA / postgresql.conf`

Log:
```25.11.2017 19:25:38AWS_ACCESS_KEY_ID is required for Wal-E but not set. Skipping Wal-E setup.
25.11.2017 19:25:38LOG:  invalid value for parameter "log_timezone": "localtime"
25.11.2017 19:25:38LOG:  invalid value for parameter "TimeZone": "localtime"
25.11.2017 19:25:38FATAL:  configuration file "/var/lib/postgresql/data/postgresql.conf" contains errors```

Reply:
delete ?
this

`sed -i "s/log_timezone =.*$/log_timezone = $DEFAULT_TIMEZONE/g" $PGDATA/postgresql.conf`
`sed -i "s/timezone =.*$/timezone = $DEFAULT_TIMEZONE/g" $PGDATA/postgresql.conf`

without it

`echo "log_timezone = $ DEFAULT_TIMEZONE" >> $ PGDATA / postgresql.conf`
`echo "timezone = $ DEFAULT_TIMEZONE" >> $ PGDATA / postgresql.conf`

does not work

Reply:
I think the correct workaround can be:
- check (using `grep`) if `timezone =` and `log_timezone =` are already present in the file
- if yes, use `sed` to set the value
- if not, append the configuration lines using `echo >>`

Reply:
Or you could remove the timezone and log_timezone lines first and then append them:
```
sed -i "s/log_timezone =.*$//g" $PGDATA/postgresql.conf
sed -i "s/timezone =.*$//g" $PGDATA/postgresql.conf
echo "log_timezone = $ DEFAULT_TIMEZONE" >> $ PGDATA / postgresql.conf
echo "timezone = $ DEFAULT_TIMEZONE" >> $ PGDATA / postgresql.conf
```